deployments/prod/environment.py

```diff
@@ -45,11 +45,11 @@ def env() -> Mapping[str, Optional[str]]:
         ]),
 
         'AZUL_TDR_SOURCES': ','.join([
-            'tdr:broad-datarepo-terra-prod-hca2:snapshot/hca_prod_20201118_dcp1___20201209:',
+            'tdr:broad-datarepo-terra-prod-hca2:snapshot/hca_prod_20201118_dcp1___20201209::2',
```

Comment:
```suggestion
            'tdr:broad-datarepo-terra-prod-hca2:snapshot/hca_prod_20201118_dcp1___20201209:-2',
```

So that both prefix-related settings are in the same field. `-` is not allowed to occur in a UUID prefix so we can use it as a delimiter. `…:aa-1` means use `aa` for prefix and a partition prefix of length one. `…:aa` means to use prefix `aa` and the default prefix length.  `…:aa-` is not allowed. `…:` means to use no prefix and the default partition prefix length.

The ticket specifies 

> I don't think that there is a reasonable default that works for all projects so we shouldn't hard-code one anywhere. 

but the addition of `AZUl_PARTITION_PREFIX_LENGTH` goes against that. That had me rethink this.

Reply:
And since 2 is actually the default, we shouldn't have to duplicate it here:

```suggestion
            'tdr:broad-datarepo-terra-prod-hca2:snapshot/hca_prod_20201118_dcp1___20201209:',
```

Reply:
>So that both prefix-related settings are in the same field. - is not allowed to occur in a UUID prefix so we can use it as a delimiter.

someone COULD choose to use `6cab9e80-0` as a prefix, maybe we should use an alternative delimiter, how about `~`?

Reply:
Valid point. I thought I wrote code to only allow hex characters but I can't find it. So let's use `/`

Independently, we should disallow prefixes that end in a dash. I don't like the ambiguity: `6cab9e80-` and `6cab9e80` match the same set of UUIDs.